examples/greeter-extension/src/types.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/example-greeter-extension
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  BindingScope,
+  BindingTemplate,
+  filterByTag,
+  inject,
+} from '@loopback/context';
+
+/**
+ * Typically an extension point defines an interface as the contract for
+ * extensions to implement
+ */
+export interface Greeter {
+  language: string;
+  greet(name: string): string;
+}
+
+/**
+ * A factory function to create binding template for extensions of the given
+ * extension point
+ * @param extensionPoint Name/id of the extension point
+ */
+export function extensionFor(extensionPoint: string): BindingTemplate {
+  return binding =>
+    binding.inScope(BindingScope.SINGLETON).tag({extensionPoint});
+}
+
+/**
+ * A binding template for greeter extensions
+ */
+export const asGreeter: BindingTemplate = extensionFor('greeter');
+
+/**
+ * Shortcut to inject extensions for the given extension point. To be promoted
+ * as `@extensions` in `@loopback/core` module.
+ *
+ * @param extensionPoint Name/id of the extension point
+ */
+export function extensions(extensionPoint: string) {
```

Comment:
Perhaps `injectExtensions` to be promoted to `@inject.extensions` may be a better name?

Reply:
Yes. Once we are settled on the final approach.